src/action/finder/download.rs

```diff
@@ -0,0 +1,295 @@
+
+        assert_eq!(
+            chunks.next().unwrap().unwrap(),
+            "some_".bytes().collect::<Vec<_>>()
+        );
+        assert!(chunks.next().is_none());
+    }
+
+    #[test]
+    fn test_skip_above_max_size() {
+        let tempdir = tempfile::tempdir().unwrap();
+        let path = tempdir.path().join("f");
+        std::fs::write(&path, "some_1").unwrap();
+        let entry = Entry {
+            metadata: path.metadata().unwrap(),
+            path,
+        };
+
+        let result = download(
+            &entry,
+            &DownloadActionOptions {
+                max_size: 5,
+                oversized_file_policy: DownloadOversizedFilePolicy::Skip,
+                use_external_stores: false,
+                chunk_size: 5,
+            },
+        );
+
+        assert!(matches!(result, Response::Skip()));
+    }
+
+    #[test]
+    fn test_hash_above_max_size() {
+        let tempdir = tempfile::tempdir().unwrap();
+        let path = tempdir.path().join("f");
+        std::fs::write(&path, "some_1").unwrap();
+        let entry = Entry {
+            metadata: path.metadata().unwrap(),
+            path,
+        };
+
+        let result = download(
+            &entry,
+            &DownloadActionOptions {
+                max_size: 5,
+                oversized_file_policy:
+                    DownloadOversizedFilePolicy::HashTruncated,
+                use_external_stores: false,
+                chunk_size: 5,
+            },
+        );
+
+        assert!(matches!(
+            result,
+            Response::HashRequest(HashActionOptions {
+                max_size: 5,
+                oversized_file_policy: HashOversizedFilePolicy::HashTruncated,
+            })
+        ));
+    }
+
+    #[test]
+    fn test_download_truncated_above_max_size() {
+        let tempdir = tempfile::tempdir().unwrap();
+        let path = tempdir.path().join("f");
+        std::fs::write(&path, "some_1").unwrap();
+        let entry = Entry {
+            metadata: path.metadata().unwrap(),
+            path,
+        };
+
+        let result = download(
+            &entry,
+            &DownloadActionOptions {
+                max_size: 5,
+                oversized_file_policy:
+                    DownloadOversizedFilePolicy::DownloadTruncated,
+                use_external_stores: false,
+                chunk_size: 3,
+            },
+        );
+
+        let mut chunks = match result {
+            Response::CollectData(chunks) => chunks,
+            _ => panic!("Unexpected result type."),
+        };
+
+        assert_eq!(
+            chunks.next().unwrap().unwrap(),
+            "som".bytes().collect::<Vec<_>>()
+        );
+        assert_eq!(
+            chunks.next().unwrap().unwrap(),
+            "e_".bytes().collect::<Vec<_>>()
+        );
+        assert!(chunks.next().is_none());
+    }
+
+    #[test]
+    fn test_chunk_id() {
+        let chunk = ChunkId::make(&"some_test_data".as_bytes().to_vec(), 5);
```

Comment:
If you apply the suggestion with using slices, then you can use bytestring literal here.

Reply:
Done!